### PR TITLE
Use reqwest to download libsodium sources

### PR DESCRIPTION
**Pull request by Fraser999**
This incorporates #51 by @dbrgn with some minor updates.

### CODE REVIEW

File: rust_sodium-sys/build.rs [hunk 1]
```diff
@@ -47,31 +47,31 @@ fn main() {
...
+#[cfg(not(feature = "use-installed-libsodium"))]
+use std::io::Cursor;
+
 /// Download the specified URL into the specified target.
 ///
 /// If something fails, an error message string is returned.
 #[cfg(not(feature = "use-installed-libsodium"))]
-fn download(url: &str, target: &std::path::Path) -> Result<(), String> {
-    use std::fs::File;
+fn download(url: &str) -> Result<Cursor<Vec<u8>>, String> {
```
Comment (dbrgn) [inline rust_sodium-sys/build.rs]: Nice, this is even better :)

**Review by dbrgn — APPROVED** [review]

File: rust_sodium-sys/build.rs [hunk 2]
```diff
@@ -54,24 +54,21 @@ use std::io::Cursor;
 ///
 /// If something fails, an error message string is returned.
```
Comment (dbrgn) [inline rust_sodium-sys/build.rs]: That comment is probably now outdated :)

Comment (Fraser999) [reply rust_sodium-sys/build.rs]: Fixed thanks!

**Review by ustulation — APPROVED** [review]

**Comment by Fraser999** [comment]
@dbrgn Are you in a rush to get this change published to crates.io?  If not, I'll try and pick off a few more issues before submitting a version change PR and publishing the updates.

**Comment by dbrgn** [comment]
@Fraser999 no, since it doesn't do any functional change there's no hurry :)

**Comment by dbrgn** [comment]
#52 would be awesome to have in the next release though 😉 That would also get rid of #16.

**Comment by Fraser999** [comment]
@dbrgn Cool - thanks.  And hint received loud and clear 😄

**Pull request merged**